fix: instrument guide

rahul31124 opened the pull request:
Fixes #3011 

## Changes
- Enabled the instrument guide to be dragged/expanded to the full height of the screen.
- Fixed layout issues to allow simultaneous zooming and scrolling of the guide content.
- Updated the Wave Generator guide images.

## Screenshots / Recordings

https://github.com/user-attachments/assets/d71db857-06a8-4b20-844b-41056180d141



**Checklist**: <!-- Please tick following check boxes with `[x]` if the respective task is completed -->
- [x] **No hard coding**: I have used values from `constants.dart` or localization files instead of hard-coded values.
- [x] **No end of file edits**: No modifications done at end of resource files.
- [x] **Code reformatting**: I have formatted the code using `dart format` or the IDE formatter.
- [x] **Code analysis**: My code passes checks run in `flutter analyze` and tests run in `flutter test`.

## Summary by Sourcery

Improve the instrument guide drawer’s usability and update wave generator assets.

New Features:
- Allow the instrument guide drawer to be resized via drag gestures up to nearly full-screen height.

Bug Fixes:
- Resolve interaction issues so guide content can be zoomed and scrolled simultaneously without layout conflicts.
- Prevent guide images from intercepting pointer events so zooming and scrolling work reliably.

Enhancements:
- Refine the guide drawer presentation with a transparent backdrop, shadow, and dynamic header behavior that toggles between expanded and collapsed states.
- Simplify stateless guide text widgets’ constructors for cleaner usage.

Documentation:
- Update wave generator circuit image assets used in the wave generator screen to the latest versions.

rahul31124 committed:
guide_issue_fixes

rahul31124 committed:
typo

rahul31124 committed:
correct condition

rahul31124 commented:
I think `test_integration/screenshots.dart` needs to be updated

CloudyPadmal commented on lib/view/widgets/guide_widget.dart:
```diff
@@ -66,103 +82,122 @@ class _InstrumentOverviewDrawerState extends State<InstrumentOverviewDrawer>
+                        onTap: () {
+                          setState(() {
+                            _currentHeightFactor =
+                                _currentHeightFactor < 0.5 ? 0.8 : 0.15;
```
In `L30` the upper bound is _0.9_. Here it is _0.8_. It is best to define these magic numbers in a constant file and refer instead.

CloudyPadmal commented on lib/view/widgets/guide_widget.dart:
```diff
@@ -66,103 +82,122 @@ class _InstrumentOverviewDrawerState extends State<InstrumentOverviewDrawer>
+                            mainAxisAlignment: MainAxisAlignment.center,
+                            children: [
+                              Icon(
+                                _currentHeightFactor > 0.5
```
I see in some places the `_currentHeightFactor` is compared strictly against 0.5 and in some places weakly. Better use the same convention to avoid corner cases.

rahul31124 committed:
corrected logic

rahul31124 commented:
Hello @CloudyPadmal,
I have addressed the requested changes and corrected the guide drawer logic that was causing the screenshot build to fail. Previously, tapping Hide did not actually close the drawer (it could only be dismissed by dragging). This has now been fixed so that pressing Hide properly closes the drawer, allowing the screenshot tests to behave as expected
